### PR TITLE
Rewrite await logic for waiting for a build

### PR DESCRIPTION
**Pull request by fridex**
## This introduces a breaking change

- [x] No

### CODE REVIEW

File: openshift/moldavite-jupyterbook-build.yaml [hunk 1]
```diff
@@ -164,15 +164,33 @@ objects:
...
+        source: |-
+          set -x
+          while true
+          do
+            state=$(
+              kubectl get pipelineruns -o go-template \
+              --template '{{range .status.conditions}}{{.reason}}{{" "}}{{"\n"}}{{end}}' \
+              "${MOLDAVITE_BOOK_ID}" \
+            )
+            [[ $? -eq 0 ]] || exit 1
```
Comment (harshad16) [inline openshift/moldavite-jupyterbook-build.yaml:181]: would this true, even if the condition is True, as exit would be 0, on get the output itself.

Comment (fridex) [reply openshift/moldavite-jupyterbook-build.yaml:181]: The check will makes sure the `kubectl` command exists with zero exit status (the state is successfully obtained), If that's not the case, the task will fail (ex. when `MOLDAVITE_BOOK_ID` is accidentally not found).

Comment (harshad16) [reply openshift/moldavite-jupyterbook-build.yaml:181]: thanks for explanation.(my eyes noticed || as & :sweat: :see_no_evil: )

**Review by harshad16 — APPROVED** [review]
thanks :100:

**Pull request merged**